### PR TITLE
[GTK4 Prep] Update FileView item context menus now created using GLib.MenuModel

### LINKED ISSUE #1157: Code

### PR DESCRIPTION
**Pull request by colinkiama**
Final part of "Replace Gtk.MenuItem with GLib.MenuModel" task in #1157

### CODE REVIEW

File: src/Utils.vala [hunk 1]
```diff
@@ -180,4 +180,112 @@ namespace Scratch.Utils {
...
+            if (app_info.get_id () == app_id) {
+                try {
+                    app_info.launch (file_list, null);
+                    print ("Launching app!\n");
+                } catch (Error e) {
+                    warning (e.message);
+                }
+                break;
+            }
+        }
```
Comment (danirabbit) [inline src/Utils.vala:242]: Can you explain why you're getting all apps for a file type etc here and not just creating a new https://valadoc.org/gio-unix-2.0/GLib.DesktopAppInfo.html ?

File: src/FolderManager/FileItem.vala [hunk 2]
```diff
@@ -18,7 +18,7 @@
  * Authored by: Julien Spautz <spautz.julien@gmail.com>, Andrei-Costin Zisu <matzipan@gmail.com>
  */
 
-namespace Scratch.FolderManager {
+ namespace Scratch.FolderManager {
```
Comment (danirabbit) [inline src/FolderManager/FileItem.vala:21]: extra space here

File: src/FolderManager/FileItem.vala [hunk 3]
```diff
@@ -73,92 +52,72 @@ namespace Scratch.FolderManager {
...
+                GLib.Action.print_detailed_name (
+                    FileView.ACTION_PREFIX + FileView.ACTION_SHOW_APP_CHOOSER,
+                    file.path
+                )
+            );
+            open_in_extra_section.append_item (open_in_other_menu_item);
+
+            open_in_menu.append_section (null, open_in_top_section);
+            open_in_menu.append_section (null, open_in_app_section);
+            open_in_menu.append_section (null, open_in_extra_section);
```
Comment (danirabbit) [inline src/FolderManager/FileItem.vala:100]: This is kind of messy:
```suggestion

            var open_in_top_section = new GLib.Menu ();
            open_in_top_section.append_item (new_window_menu_item);

            var open_in_app_section = Utils.create_executable_app_items_for_file (file.file, file_type);

            var open_in_other_menu_item = new GLib.MenuItem (
                _("Other Application…"),
                GLib.Action.print_detailed_name (
                    FileView.ACTION_PREFIX + FileView.ACTION_SHOW_APP_CHOOSER,
                    file.path
                )
            );

            var open_in_extra_section = new GLib.Menu ();
            open_in_extra_section.append_item (open_in_other_menu_item);

            var open_in_menu = new GLib.Menu ();
            open_in_menu.append_section (null, open_in_top_section);
            open_in_menu.append_section (null, open_in_app_section);
            open_in_menu.append_section (null, open_in_extra_section);
```

File: src/FolderManager/FolderItem.vala [hunk 4]
```diff
@@ -88,147 +88,116 @@ namespace Scratch.FolderManager {
...
-            if (info != null) {
-                List<AppInfo> external_apps = GLib.AppInfo.get_all_for_type (file_type);
+            var file_type = info.get_attribute_string (GLib.FileAttribute.STANDARD_CONTENT_TYPE) ?? "inode/directory";
+            var launch_app_action = Utils.action_from_group (FileView.ACTION_LAUNCH_APP_WITH_FILE_PATH,
+                                                             view.actions) as SimpleAction;
+            launch_app_action.change_state (new GLib.Variant.string (file_type));
+
+            var rename_menu_item = new GLib.MenuItem (_("Rename"),
+                                                      FileView.ACTION_PREFIX
+                                                      + FileView.ACTION_RENAME_FOLDER);
```
Comment (danirabbit) [inline src/FolderManager/FolderItem.vala:115]: Code style:
```suggestion
            var rename_menu_item = new GLib.MenuItem (
                _("Rename"),
                FileView.ACTION_PREFIX + FileView.ACTION_RENAME_FOLDER
            );
```

File: src/FolderManager/ProjectFolderItem.vala [hunk 5]
```diff
@@ -67,18 +68,27 @@ namespace Scratch.FolderManager {
...
+                    GLib.VariantType.STRING,
+                    ""
+                );
                 monitored_repo.branch_changed.connect (branch_or_name_changed);
                 monitored_repo.ignored_changed.connect ((deprioritize_git_ignored));
                 monitored_repo.file_status_change.connect (() => update_item_status (null));
                 monitored_repo.update_status_map ();
                 monitored_repo.branch_changed ();
+                change_branch_action.activate.connect (handle_change_branch_action);
+
```
Comment (danirabbit) [inline src/FolderManager/ProjectFolderItem.vala:91]: extra whitespace

**Review by danirabbit — CHANGES_REQUESTED** [review]
This is a lot to review at one time. It's kind of difficult. Do you think you could break this up a bit? Maybe first a branch with some of these changes to existing actions like making the branch action stateful, a branch changing some of these classes to use actions like for example adding actions to FileView could honestly be its own branch, and then finally a branch changing things to use glib menu at the end of that? There's just so many changes here that it's hard to see what's going on and there's quite a few code style and white space errors etc

File: src/Utils.vala (hunk 1 above)
Comment (colinkiama) [reply src/Utils.vala:242]: Honestly, I can't remember. I did this months ago. Creating a new `DesktopAppInfo` instance is much more efficient. Not sure how I missed that.

**Comment by danirabbit** [comment]
@colinkiama did you want to try to resolve conflicts here or should this be closed?

**Comment by colinkiama** [comment]
This should be closed.

**Pull request closed without merging**